Unbreak haskellPackages.dhall-nix

KAction opened the pull request:
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

KAction committed:
Unbreak haskellPackages.dhall-nix

maralorn commented:
Halle @KAction thank you for your PR!

Sadly this PR can‘t work this way. The haskell-packages.nix file is autogenerated by the hackage2nix tool from the configuration-hackage2nix.yaml. So your changes would be overriden in the night after a merge.

So to fix this you need to do 2 things.
1. The overrides of the dependency versions belong in configuration-common.nix. There are plenty of examples there.
2. To mark a package unbroken please remove it from the `broken` list in the yaml config mentioned above.

KAction commented:
Thank you for explanation.

I updated `dhall-nix` override in `configuration-common.nix` and removed `dhall-nix` from list of broken packages, yet it is still marked as broken and `nix-build` refuses to build it, unless forced. If I force it

````
$ NIXPKGS_ALLOW_BROKEN=1 nix-build -A dhall-nix
````

it builds fine. Can you please instruct what else should I do?

cdepillabout commented on pkgs/development/haskell-modules/configuration-common.nix:
```diff
@@ -920,7 +920,14 @@ self: super: {
   dhall-json = generateOptparseApplicativeCompletions ["dhall-to-json" "dhall-to-yaml"] super.dhall-json;
-  dhall-nix = generateOptparseApplicativeCompletion "dhall-to-nix" (super.dhall-nix);
+  dhall-nix = generateOptparseApplicativeCompletion "dhall-to-nix" (
+    super.dhall-nix.override {
+      dhall = super.dhall_1_34_0.override {
+        repline = self.repline_0_4_0_0.override {
+          haskeline = self.haskeline_0_8_0_0;
+        };
+      };
```
You could probably rewrite this a little cleaner with `overrideScope` (instead of a normal `override`). There should be a couple uses of it in this file to copy.

KAction replied:
Ok, done.

cdepillabout commented on pkgs/development/haskell-modules/configuration-hackage2nix.yaml:
```diff
@@ -4415,7 +4415,6 @@ broken-packages:
   - dhall-check
   - dhall-docs
   - dhall-fly
-  - dhall-nix
```
Since you've removed this line, the next time the automatic Hackage updater script runs, it will mark `dhall-nix` as unbroken.

You shouldn't have to do anything on your end, as long as you've checked it does actually compile correctly when enabling building packages marked broken.

cdepillabout reviewed (changes requested):
This looks good, but one last request.

Can you target the `haskell-updates` branch instead of `master`?  We generally like Haskell-related PRs to go into the `haskell-updates` branch instead of `master`.

KAction commented:
Ok, done.

cdepillabout commented:
I've confirmed that `dhall-nix` compiles correctly.  Thanks!
